flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java

```diff
@@ -412,7 +412,7 @@ public void registerType(Class<?> type) {
 		if (from > to) {
 			throw new IllegalArgumentException("Start of sequence must not be greater than the end");
 		}
-		return fromCollection(new NumberSequenceIterator(from, to), BasicTypeInfo.LONG_TYPE_INFO, "Sequence Source");
+		return addSource(new StatefulSequenceSource(from, to), "Sequence Source").setParallelism(1);
 	}
```

Comment:
But why wouldn't we allow the user to explicitly set a higher parallelism. The call is quite explicit and I think I would be more surprised as a user if my source did not run in parallel even though I set a higher parallelism.